### PR TITLE
[LM-83 LM-84] add basic liquidity-manager test setup

### PR DESCRIPTION
**Pull request by anirudhwarrier**
## Motivation

Liquidity Manager doesn't have any end to end smoke tests. 

## Solution

Add a most basic LM smoke test.
- The tests are very basic atm and do nothing more than is absolutely necessary of it.
    - More features will be added to the test setup as more test cases are implemented (after test cases are listed down first)
    - For now this works only on the simulated/ private geth networks - 1337 and 2337
- This PR doesn’t add this to CI. That will be a separate PR
- There are some TODOs that I have added in the tests, which I plan to tackle in next steps after this PR is merged.

### CODE REVIEW

**Comment by anirudhwarrier** [comment]
To run the tests
1. `cd integration-tests`
2. `make build_docker_image image=chainlink tag=ccip-develop`
3. `cd ccip-tests`
4. `touch ./testconfig/override/secret.toml`
5. `vi ./testconfig/override/secret.toml`
```[CCIP]
[CCIP.Env]

# ChainlinkImage is mandatory for all tests.
[CCIP.Env.NewCLCluster]
[CCIP.Env.NewCLCluster.Common]
[CCIP.Env.NewCLCluster.Common.ChainlinkImage]
image = "chainlink"
version = "ccip-develop"

[CCIP.Env.Logging]
test_log_collect=true
```
6. `export BASE64_CCIP_SECRETS_CONFIG=$(base64 -i ./testconfig/override/secret.toml)`
7. `export BASE64_CCIP_CONFIG_OVERRIDE=$(base64 -i ./testconfig/tomls/ccip-default.toml)`
8. `go test -v -timeout 60m -count=1 -test.parallel=1 -run ^TestLmBasic ./smoke`

File: integration-tests/ccip-tests/smoke/lm_test.go [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+	l1ChainId := lmTestSetup.Cfg.SelectedNetworks[0].ChainID
+	l2ChainId := lmTestSetup.Cfg.SelectedNetworks[1].ChainID
+
+	l1liquidityStart, err := lmTestSetup.LMModules[l1ChainId].LM.GetLiquidity()
+	require.NoError(t, err, "Failed to get liquidity from L1")
+	l2liquidityStart, err := lmTestSetup.LMModules[l2ChainId].LM.GetLiquidity()
+	require.NoError(t, err, "Failed to get liquidity from L2")
+	log.Info().Str("L1 Liquidity", l1liquidityStart.String()).Str("L2 Liquidity", l2liquidityStart.String()).Msg("Liquidity at start")
+
+	//TODO: Improve this wait
```
Comment (amirylm) [inline integration-tests/ccip-tests/smoke/lm_test.go:31]: Could you elaborate? this is a point where we'll need to read logs to find our events

Comment (anirudhwarrier) [reply integration-tests/ccip-tests/smoke/lm_test.go:31]: Yes. So right now this step is stupid simple and waits for 3 minutes. And in this time span ~13 OCR rounds happen and the LM does some rebalancing in this time. And to confirm that there was some rebalancing done, the liquidity before and after the test is compared and if there is a difference, then that is LM doing its job. Else the test fails. Have added a [TODO](https://github.com/smartcontractkit/ccip/pull/877/files#diff-0f9dfbde2374e47c52de5e17a9f7797927ccc5484f9665a4c109c17fa4aa9b49R47) down below to make the test to watch for LiquidityTransferred event on chain.

**Review by amirylm — APPROVED** [review]
LGTM

We should consider exposing some CLI (or modify existing) for `integration-tests/ccip-tests/testsetups/lm.go`, it can be used to spin up local env

**Pull request merged**